### PR TITLE
adjust new realm for PocCreation

### PR DESCRIPTION
**Pull request by Hiroki6**

### CODE REVIEW

File: src/main/resources/db/migration/V11_0__add_poc_tenant_type_group_id.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE poc_manager.poc_table
```
Comment (Hiroki6) [inline src/main/resources/db/migration/V11_0__add_poc_tenant_type_group_id.sql:1]: I will fix this versioning

File: src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala [hunk 2]
```diff
@@ -6,6 +6,7 @@ import com.ubirch.models.keycloak.group.{ GroupCreationError, GroupId, GroupName
 import com.ubirch.models.keycloak.roles.{ CreateKeycloakRole, RoleAlreadyExists, RoleCreationException, RoleName }
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala:6]: This file was mainly changed

File: src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala [hunk 3]
```diff
@@ -120,19 +139,46 @@ class KeycloakHelperImpl @Inject() (
...
 
-  override def createEmployeeGroup(pocAndStatus: PocAndStatus): Task[PocAndStatus] = {
+  override def createPocTenantTypeGroup(pocAndStatus: PocAndStatus, tenant: Tenant): Task[PocAndStatus] = {
+    if (pocAndStatus.status.pocTenantTypeGroupCreated.isEmpty || pocAndStatus.status.pocTenantTypeGroupCreated.contains(
+        true))
+      Task(pocAndStatus)
+    else {
+      addSubGroup(
+        tenant.certifyGroupId.value,
+        pocAndStatus.poc.roleName,
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala:163]: This roleName can be used for the name of pocTenantTypeGroup?

File: src/main/resources/db/migration/V11_0__add_poc_tenant_type_group_id.sql (hunk 1 above)
Comment (lalottacontinua) [reply src/main/resources/db/migration/V11_0__add_poc_tenant_type_group_id.sql:1]: or just let's merge all V11 in the feature/use-multiple-.. branch.

File: src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala (hunk 3 above)
Comment (lalottacontinua) [reply src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala:163]: yes.

File: src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala [hunk 4]
```diff
@@ -120,19 +139,46 @@ class KeycloakHelperImpl @Inject() (
...
+        CertifyKeycloak)
+        .map { groupId =>
+          val updatedPoc = pocAndStatus.poc.copy(pocTenantTypeGroupId = Some(groupId.value))
+          val updatedStatus = pocAndStatus.status.copy(pocTenantTypeGroupCreated = Some(true))
+          pocAndStatus.copy(poc = updatedPoc, status = updatedStatus)
+        }
+    }
+  }
+
+  override def createEmployeeGroup(pocAndStatus: PocAndStatus, tenant: Tenant): Task[PocAndStatus] = {
```
Comment (lalottacontinua) [inline src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala:175]: here you wouldn't need the tenant. if you merge my PR you can just call poc.getRealm.

File: src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala [hunk 5]
```diff
@@ -141,15 +187,15 @@ class KeycloakHelperImpl @Inject() (
     }
   }
 
-  override def assignEmployeeRole(pocAndStatus: PocAndStatus): Task[PocAndStatus] = {
+  override def assignEmployeeRole(pocAndStatus: PocAndStatus, tenant: Tenant): Task[PocAndStatus] = {
```
Comment (lalottacontinua) [inline src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala:190]: same here.

File: src/main/scala/com/ubirch/models/poc/PocStatus.scala [hunk 6]
```diff
@@ -72,6 +73,7 @@ object PocStatus {
       clientCertProvided = if (poc.clientCertRequired) Some(false) else None,
       adminGroupCreated = if (poc.certifyApp) Some(false) else None,
       adminRoleAssigned = if (poc.certifyApp) Some(false) else None,
+      pocTenantTypeGroupCreated = if (poc.certifyApp) Some(false) else None,
```
Comment (lalottacontinua) [inline src/main/scala/com/ubirch/models/poc/PocStatus.scala:76]: we could rename it pocTypeGroup...

**Review by lalottacontinua — COMMENTED** [review]
just a few comments.

File: src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala [hunk 7]
```diff
@@ -120,19 +139,46 @@ class KeycloakHelperImpl @Inject() (
...
   }
 
-  override def createEmployeeGroup(pocAndStatus: PocAndStatus): Task[PocAndStatus] = {
+  override def createPocTenantTypeGroup(pocAndStatus: PocAndStatus, tenant: Tenant): Task[PocAndStatus] = {
+    if (pocAndStatus.status.pocTenantTypeGroupCreated.isEmpty || pocAndStatus.status.pocTenantTypeGroupCreated.contains(
+        true))
+      Task(pocAndStatus)
+    else {
+      addSubGroup(
+        tenant.certifyGroupId.value,
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/KeycloakHelper.scala:162]: I took mistake. Here should be `tenantTypeGroupId`. will fix it

**Pull request merged**